dev/NavigationView/NavigationView.cpp

```diff
@@ -1863,10 +1943,7 @@ void NavigationView::KeyboardFocusLastItemFromItem(const winrt::NavigationViewIt
         auto lastIndex = itemsSourceView.Count() - 1;
         if (auto lastElement = ir.TryGetElement(lastIndex))
         {
-            if (auto controlLast = lastElement.try_as<winrt::Control>())
-            {
-                controlLast.Focus(winrt::FocusState::Programmatic);
-            }
+            SetFocus(lastElement, winrt::FocusState::Keyboard);
```

Comment:
>Keyboard [](start = 53, length = 8)

why change from programmatic?

Reply:
Changing the focus using programmatic was causing issues for me where if I was changing the focus to an item that was offscreen, it would not scroll to the newly focused item.

---
In reply to: [367553471](https://github.com/microsoft/microsoft-ui-xaml/pull/1850#discussion_r367553471) [](ancestors = 367553471)

Reply:
Interesting, is the last used interaction device guaranteed to be keyboard when we are in this function?